### ISSUE TITLE
Conformer : Training ASR for multiple language 

### ISSUE THREAD

**Issue by mujhenahiata** [NVIDIA/NeMo#5942]
**What i did**

I took a multilingual SSL pretrained model and wanted to fine-tune it for ASR task on english and hindi together at once. I used a tokenizer that contains both English and Hindi manifest. 

**ERROR** 
After few steps the loss goes nan , detected inf or nan values in gradients! Setting gradients to zero.

**What i tried**
All of the hyper-parameters are kept as mentioned in GitHub repo.
i tried different combination of batch size and gradient accumulation. i tried lowering the learning rate.
____________________________________________________________
**AIM of my experiment**
let say the audio clip says : Aj mene apple khaya
want the transcription to be : आज मेने apple खाया
____________________________________________________________
how can this be achieved. 
Do i need to use the other method of tokenizing , where one make separate tokenizer for languages, and  use "agg" aggregate.
Can you suggest any method to achieve the above problem

An early response is always appreciated.
@titu1994 @meghmak13

**Comment by titu1994** [comment]
Are you using a CTC or RNNT model ? CTC might fail if your audio is too short compared to your text. Also what is the size of the Tokenizer ? You should try with 1024 tokens first

**Comment by mujhenahiata** [comment]
i am using CTC, i read that 1024 if using RNNT

**Comment by titu1994** [comment]
CTC can and should have more tokens, at least 1024. 256 is not enough for two languages possible

**Comment by mujhenahiata** [comment]
Ok. What is your view on metrics , that I would consider , val loss , wer or cer.

**Comment by mujhenahiata** [comment]
Can this https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/conformer/multilang/conformer_ctc_bpe_multilang.yaml
Setting give results like this if yes then how. I was not able to find one.
let say the audio clip says : Aj mene apple khaya
want the transcription to be : आज मेने apple खाया
How one can do code switching?

**Comment by titu1994** [comment]
We have tutorial for multilingual and code switched ASR - https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Multilang_ASR.ipynb

Plus here is a post by @bmwshop that explains how multilingual and code switched ASR works in NeMo - https://developer.nvidia.com/blog/multilingual-and-code-switched-automatic-speech-recognition-with-nvidia-nemo/
